### PR TITLE
Rename to maplibre-gl-geocoder, change to a pluggable geocoder api

### PR DESCRIPTION
**Pull request by thaddmt**
#### Overview

- Working on removing mapbox dependencies (mapbox sdk, events service sdk)
- Changed to a pluggable geocoder API that must be passed into mapbox-gl-geocoder
- Working on updating unit tests, but pretty much all unit tests need to be updated since they previously called the mapbox api in test, changing to use mocked data

#### Checklist
 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] run `npm run docs` and commit changes to API.md
 - [ ] update CHANGELOG.md with changes under `master` heading before merging

### CODE REVIEW

**Comment by Amplifiyer** [comment]
This PR is very hard to review due to formatting changes, is it possible to separate these?

**Comment by thaddmt** [comment]
> This PR is very hard to review due to formatting changes, is it possible to separate these?

@Amplifiyer updated to remove formatting changes!

**Comment by Amplifiyer** [comment]
It looks like we should keep both maplibre and mapbox as CSS names to avoid breaking changes for customers overriding these classes? Ref: https://github.com/maplibre/maplibre-gl-js/pull/203/files

**Comment by thaddmt** [comment]
> It looks like we should keep both maplibre and mapbox as CSS names to avoid breaking changes for customers overriding these classes? Ref: https://github.com/maplibre/maplibre-gl-js/pull/203/files

Sure can add them back in

**Review by TreTuna — APPROVED** [review]

File: API.md [hunk 1]
```diff
@@ -49,149 +49,149 @@
...
+    -   `options.minLength` **[Number][58]** Minimum number of characters to enter before results are shown. (optional, default `2`)
+    -   `options.limit` **[Number][58]** Maximum number of results to show. (optional, default `5`)
+    -   `options.language` **[string][62]?** Specify the language to use for response text and query result weighting. Options are IETF language tags comprised of a mandatory ISO 639-1 language code and optionally one or more IETF subtags for country or script. More than one value can also be specified, separated by commas. Defaults to the browser's language settings.
+    -   `options.filter` **[Function][65]?** A function which accepts a Feature in the [Carmen GeoJSON][66] format to filter out results from the Geocoding API response before they are included in the suggestions list. Return `true` to keep the item, `false` otherwise.
+    -   `options.localGeocoder` **[Function][65]?** A function accepting the query string which performs local geocoding to supplement results from the Maplibre Geocoding API. Expected to return an Array of GeoJSON Features in the [Carmen GeoJSON][66] format.
```
Comment (Amplifiyer) [inline API.md:93]:
```suggestion
    -   `options.localGeocoder` **[Function][65]?** A function accepting the query string which performs local geocoding to supplement results from the Geocoding API. Expected to return an Array of GeoJSON Features in the [Carmen GeoJSON][66] format.
```

File: API.md [hunk 2]
```diff
@@ -49,149 +49,149 @@
...
+    -   `options.externalGeocoder` **[Function][65]?** A function accepting the query string, current features list, and geocoder options which performs geocoding to supplement results from the Maplibre Geocoding API. Expected to return a Promise which resolves to an Array of GeoJSON Features in the [Carmen GeoJSON][66] format.
```
Comment (Amplifiyer) [inline API.md:94]:
```suggestion
    -   `options.externalGeocoder` **[Function][65]?** A function accepting the query string, current features list, and geocoder options which performs geocoding to supplement results from the Geocoding API. Expected to return a Promise which resolves to an Array of GeoJSON Features in the [Carmen GeoJSON][66] format.
```

File: package.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
   ],
   "repository": {
     "type": "git",
-    "url": "git+ssh://git@github.com/mapbox/mapbox-gl-geocoder.git"
+    "url": "git+ssh://git@github.com/thaddmt/maplibre-gl-geocoder.git"
```
Comment (Amplifiyer) [inline package.json:23]: we should change that to maplibre

**Review by Amplifiyer — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (thaddmt) [reply package.json:23]: Was planning to change it after we actually move it

**Pull request merged**